grails-app/services/com/unifina/service/PermissionService.groovy

```diff
@@ -121,6 +144,34 @@ class PermissionService {
 		(Operation.CANVAS_STARTSTOP): [Operation.STREAM_PUBLISH],
 	]
 
+	private Permission hasTransitiveDashboardPermissions(Canvas canvas, Userish userish) {
+		/*
+		List<DashboardItem> items = DashboardItem.findAllByCanvas(canvas)
```

Comment:
- Find out the module id by parsing `Stream#uiChannelPath`, which begins with:
`/canvases/[canvasId]/modules/[moduleId]`
- Add to this query the filter condition `eq('module', moduleId)`

Reply:
Actually looks like this is pretty much implemented currently as `StreamService#isPermissionToStreamViaDashboard`. Let's refactor this to have all the transitive permission checking happening in `PermissionService`!